temporian/implementation/numpy/operators/combine.py

```diff
@@ -0,0 +1,112 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Implementation for the Combine operator."""
+
+
+from typing import Dict, List, Set
+import numpy as np
+
+from temporian.implementation.numpy.data.event_set import (
+    IndexData,
+    EventSet,
+    IndexType,
+)
+from temporian.core.operators.combine import Combine
+from temporian.core.operators.combine import (
+    FROM_FIRST,
+    FROM_INTERSECT,
+    FROM_UNION,
+)
+from temporian.implementation.numpy import implementation_lib
+from temporian.implementation.numpy.operators.base import OperatorImplementation
+
+
+class CombineNumpyImplementation(OperatorImplementation):
+    def __init__(self, operator: Combine) -> None:
+        assert isinstance(operator, Combine)
+        super().__init__(operator)
+
+    def __call__(self, **inputs: EventSet) -> Dict[str, EventSet]:
+        assert isinstance(self.operator, Combine)
+        output_schema = self.output_schema("output")
+        output_evset = EventSet(data={}, schema=output_schema)
+        index_from = self.operator.index_from
+
+        input_evsets = list(inputs.values())
+        first_input = input_evsets[0]
+        first_features = first_input.schema.feature_names()
+
+        # What index values to use
+        if index_from == FROM_FIRST:
+            combined_keys = first_input.data.keys()
+        elif index_from == FROM_UNION:
+            combined_keys = set()
+            for evset in input_evsets:
+                combined_keys |= set(evset.data.keys())
+        elif index_from == FROM_INTERSECT:
+            combined_keys = set(first_input.data.keys())
+            for evset in input_evsets[1:]:
+                combined_keys &= set(evset.data.keys())
+        else:
+            raise ValueError(f"Unknown parameter for combine: {index_from=}")
+
+        # Concatenate timestamps and features for each index, and sort
+        for index_key in combined_keys:
+            all_timestamps: List[np.ndarray] = []
+            all_feats: Dict[str, List[np.ndarray]] = {
+                name: [] for name in first_features
+            }
+
+            # Get timestamps and features from all input EventSets
+            for evset in input_evsets:
+                if index_key not in evset.data:
+                    continue
+                index_data = evset.get_index_value(index_key, normalize=False)
+                evset_feats = evset.schema.feature_names()
+                all_timestamps.append(index_data.timestamps)
+                # Access feature data by name, since position in IndexData may vary
+                for feat in first_features:
+                    feat_idx = evset_feats.index(feat)
+                    all_feats[feat].append(index_data.features[feat_idx])
+
+            # Concatenate and sort timestamps
+            timestamps: np.ndarray = np.concatenate(all_timestamps)
```

Comment:
Is there a fonction that combines efficiently concatenate and sort?

Reply:
Couldn't find one. Maybe it's a good one to try some C++ 😄 
But I'd leave it for a future PR.

Reply:
`sortednp` lib seems to have this functionality, see https://stackoverflow.com/questions/27916710/numpy-merge-sorted-array-to-an-new-array, but I'd argue against adding the dependency just for this

Reply:
Nice one, I hadn't found it. Although I agree with not adding the dependency only for this.